internal/cloudinfo/instance_type.go

```diff
@@ -191,6 +250,10 @@ func (s *InstanceTypeService) Query(ctx context.Context, provider string, servic
 			includeInResults = includeInResults && applyNetworkCategoryFilter(product.NtwPerfCat, *query.Filter.NetworkCategory)
 		}
 
+		if query.Filter.Category != nil {
+			includeInResults = includeInResults && applyInstanceCategoryFilter(product.Category, *query.Filter.Category)
```

Comment:
In case of amazon, the `category` field currently contains values like "Compute optimized".

Reply:
these descriptions are already mapped to the generated category enums